javascript/stimulus_reflex.js

```diff
@@ -27,24 +28,15 @@ const promises = {}
 // controller - the StimulusReflex controller to subscribe
 //
 const createSubscription = controller => {
-  actionCableConsumer = actionCableConsumer || getConsumer()
+  adaptedConsumer = adaptedConsumer || new ActionCableAdapter(getConsumer())
   const { channel } = controller.StimulusReflex
   const identifier = JSON.stringify({ channel })
 
-  controller.StimulusReflex.subscription =
-    actionCableConsumer.subscriptions.findAll(identifier)[0] ||
-    actionCableConsumer.subscriptions.create(channel, {
-      received: data => {
-        if (!data.cableReady) return
-        if (data.operations.morph && data.operations.morph.length) {
-          const urls = Array.from(
-            new Set(data.operations.morph.map(m => m.stimulusReflex.url))
-          )
-          if (urls.length !== 1 || urls[0] !== location.href) return
-        }
-        CableReady.perform(data.operations)
-      }
-    })
+  let subscription =
+    adaptedConsumer.find_subscription(identifier) ||
+    adaptedConsumer.create_subscription(channel)
+
+  controller.StimulusReflex.subscription = subscription
```

Comment:
We don't really need the subscription at a later point. So we could actually delete this line.

Reply:
I think it's actually a good idea to tack it onto a global

I even also added the consumer:

```suggestion
  controller.StimulusReflex.subscription = subscription
  controller.StimulusReflex.consumer = adaptedConsumer
```

Reply:
you can then actually call it on line 61:

```
   return this.StimulusReflex.consumer.isConnected()
```

I guess you forgot that method you defined on the consumer

Reply:
I'd probably also rename `isActionCableConnectionOpen` to something more generic, `isConnectionOpen` for example